### ISSUE TITLE
Make single gradle project build execution faster 

### ISSUE THREAD

**Issue by breskeby** [elastic/elasticsearch#71797]
Running the elasticsearch build on a single project (e.g. `./gradlew :server:project`) results in all tasks of this project are executed in a serial fashion one after another without parallization as this is only supported across multiple projects. 
This is especially in our precommit builds quite ineffective as the major time of task execution is spent on a single project.

See https://gradle-enterprise.elastic.co/s/2hivxyq5naos2/timeline?details=vqapcpyehsnok&show=Details as an example that results in a `2m 37.543s` build after only one file change in :server production code

**Comment by breskeby** [comment]
raised https://github.com/gradle/gradle/pull/19361 which ended up in https://github.com/gradle/gradle/pull/20069 which will dramatically improve performance of single project builds when checkstyle is involved as single projects can now run multiple checkstyle tasks for different source sets in parallel

**Comment by pugnascotia** [comment]
@breskeby can we close this issue now?

**Comment by breskeby** [comment]
yes. The changes we provided to gradle are shipped with 7.5. closing this issue
